nexus/nexus-app/src/main/java/org/sonatype/nexus/configuration/application/DefaultNexusConfiguration.java

```
@@ -786,8 +794,10 @@ public synchronized void deleteRepository( String id )
             {
                 i.remove();
 
-                applyAndSaveConfiguration();
+                releaseRepository( repository, getConfigurationModel(), repo );
 
+                saveConfiguration();
+                
```

Well, whatever, the order should not really matter since release will swallow any plexus exception anyway.... but we can swap the order (to be resilient to some spurious runtime exceptions too?)